apps/aecontract/src/aect_state_tree.erl

```diff
@@ -51,24 +52,74 @@ insert_contract(Contract, Tree = #contract_tree{ contracts = CtTree }) ->
     Id         = aect_contracts:id(Contract),
     Serialized = aect_contracts:serialize(Contract),
     CtTree1    = aeu_mtrees:insert(Id, Serialized, CtTree),
-    Tree#contract_tree{ contracts = CtTree1 }.
+    CtTree2    = insert_store(Contract, CtTree1),
+    Tree#contract_tree{ contracts = CtTree2 }.
+
+insert_store(Contract, CtTree) ->
+    Id = aect_contracts:store_id(Contract),
+    Store = aect_contracts:state(Contract),
+    insert_store_nodes(Id, Store, CtTree).
+
+insert_store_nodes(Prefix, Store, CtTree) ->
+    Insert = fun (Key, Value, Tree) ->
+                     Id = <<Prefix/binary, Key/binary>>,
+                     aeu_mtrees:insert(Id, Value, Tree)
+             end,
+     maps:fold(Insert, CtTree, Store).
+
 
 %% @doc Update an existing contract.
 -spec enter_contract(aect_contracts:contract(), tree()) -> tree().
 enter_contract(Contract, Tree = #contract_tree{ contracts = CtTree }) ->
     Id         = aect_contracts:id(Contract),
     Serialized = aect_contracts:serialize(Contract),
     CtTree1    = aeu_mtrees:enter(Id, Serialized, CtTree),
-    Tree#contract_tree{ contracts = CtTree1 }.
+    CtTree2    = enter_store(Contract, CtTree1),
+    Tree#contract_tree{ contracts = CtTree2 }.
+
+enter_store(Contract, CtTree) ->
+    Id = aect_contracts:store_id(Contract),
+    Store = aect_contracts:state(Contract),
+    enter_store_nodes(Id, Store, CtTree).
+
+enter_store_nodes(Prefix, Store, CtTree) ->
```

Comment:
The same here as above. Actually, for the above, it might work since it is a fresh contract, but here it might fail.

Reply:
You can compute the diff here and update the trees accordingly. But you still need to explicitly delete the removed store positions.